flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -169,6 +172,11 @@ public void close() throws CatalogException {
 		}
 	}
 
+	@Override
+	public Optional<TableFactory> getTableFactory() {
+		return Optional.of(new HiveTableFactory());
```

Comment:
Does this mean `HiveTableFactory ` will create source/sink for both generic and hive tables?

Reply:
Yes. However, for generic tables, it delegates the responsibility to auto tablefactory discovery service, as you can see from HiveTableFactory implementation.